File: ruby/matrix/matrix.rb
class Matrix
  def initialize(trix)
    @trix = trix
  end
  
  def rows
    trix_rows = @trix.split("\n")
    trix_rows.map do |row|
      (row.split(" ")).map { |num| num.to_i }
    end
  end

  def columns
    # trix_columns = []
    # new_rows = rows
    # (new_rows.first.length).times do
    #   trix_columns << new_rows.map { |row| row.shift }
    # end
    # trix_columns
    rows.transpose
  end
end

File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        global matrix_list
        matrix_list = self.cleanup(matrix_string)

    def row(self, index):
        return matrix_list[index - 1]

    def column(self, index):
        list = []
        for row in matrix_list:
            list.append(row[index - 1])
        return list

    def cleanup(self, matrix_string):
        rows = matrix_string.split('\n')
        return [[int(num) for num in row.split(' ')] for row in rows]
